cmd/dep/init.go

```diff
@@ -407,7 +407,14 @@ func getProjectData(ctx *dep.Ctx, pkgT pkgtree.PackageTree, cpr string, sm gps.S
 		}
 
 		ondisk[pr] = v
-		constraints[pr] = getProjectPropertiesFromVersion(v)
+		pp := getProjectPropertiesFromVersion(v)
+		if pp.Constraint != nil || pp.Source != "" {
+			constraints[pr] = pp
+		} else {
+			if ctx.Loggers.Verbose {
+				ctx.Loggers.Err.Printf("dep: \"%v\" is unconstrained. Not adding to project data.\n", pr)
+			}
```

Comment:
This logging is not required. We are trying to remove the existing `"dep: ..."` logs at init because it makes the init output (verbose/non-verbose) confusing, combined with constraint feedback output.
The `feedback()` following this code would handle displaying that it's a revision only project constraint.